src/emulator/initEmulators.ts

```diff
@@ -0,0 +1,69 @@
+// specific initialization steps for an emulator
+
+import { promptOnce } from "../prompt";
+import { EmulatorLogger } from "./emulatorLogger";
+import { Emulators } from "./types";
+
+export const AdditionalInitFns: {
+  [e in Emulators]: () => Promise<Record<string, string> | null>;
+} = {
+  apphosting: async () => {
+    // Auto-detect package manager and set startCommandOverride
+    const logger = EmulatorLogger.forEmulator(Emulators.APPHOSTING);
+    logger.log("BULLET", "Initializing App Hosting Emulator");
+
+    const rootDirectory = await promptOnce({
+      name: "rootDir",
+      type: "input",
+      default: "./",
+      message: "Specify your app's root directory relative to your repository",
+    });
+
+    return {
+      rootDirectory,
+    };
+    // prompt for apphosting yaml to export
+  },
+  auth: async () => {
```

Comment:
Definitely no need to list out every product here with an empty function - cleaner to just include the products that actually have extra questions, and then change the call site to work with undefined